app/services/next_course_creation_step_service.rb

```diff
@@ -1,48 +1,64 @@
 class NextCourseCreationStepService
-  def execute(current_step:, current_provider:)
-    case current_step
-    when :level
-      :subjects
-    when :subjects
-      :age_range
-    when :age_range
-      :outcome
-    when :outcome
-      handle_outcome(current_provider)
-    when :fee_or_salary
-      :full_or_part_time
-    when :apprenticeship
-      :full_or_part_time
-    when :full_or_part_time
-      :location
-    when :location
-      handle_location(current_provider)
-    when :accredited_body
-      :entry_requirements
-    when :entry_requirements
-      :applications_open
-    when :applications_open
-      :start_date
-    when :start_date
-      :confirmation
-    end
+  def execute(current_step:, current_provider:, level:)
```

Comment:
How about passing `course` through instead of `level` and `provider`? I don't see why the caller should be concerned with what part of the course is needed to determine what the next step is, and we could even forseeably require some other part of the `course` one day, which would require another change to the interface here.

Reply:
I'd like to avoid giving a service much more information than it makes it harder to see what exactly it depends on.

Reply:
As discussed IRL, I think that `course` is the right level to parameterise this service on, given that it has `Course` in the name and it's specifically about finding the next step for a course.